mmv1/products/dlp/api.yaml

```diff
@@ -1328,4 +1330,267 @@ objects:
+                                    - :TUESDAY
+                                    - :WEDNESDAY
+                                    - :THURSDAY
+                                    - :FRIDAY
+                                    - :SATURDAY
+                                    - :SUNDAY
+                        - !ruby/object:Api::Type::NestedObject
+                          name: redactConfig
+                          description: |
+                            Redact a given value. For example, if used with an InfoTypeTransformation transforming PHONE_NUMBER, and input 'My phone number is 206-555-0123', the output would be 'My phone number is '.
+                          properties: [] # Meant to be an empty object with no properties - see here : https://cloud.google.com/dlp/docs/reference/rest/v2/projects.deidentifyTemplates#redactconfig
+                          # The fields below allow POST requests to include the "redactConfig" transformation in the payload
+                          # The trade-off is that POST requests specifying another transformation type will include will include "redactConfig" : null in the payload
+                          send_empty_value: true
+                          allow_empty_object: true
+                        - !ruby/object:Api::Type::NestedObject
+                          name: characterMaskConfig
+                          description: |
+                            Partially mask a string by replacing a given number of characters with a fixed character. Masking can start from the beginning or end of the string. This can be used on data of any type (numbers, longs, and so on) and when de-identifying structured data we'll attempt to preserve the original data's type. (This allows you to take a long like 123 and modify it to a string like **3.
+                          properties:
+                            - !ruby/object:Api::Type::String
+                              name: 'maskingCharacter'
+                              description: |
+                                Character to use to mask the sensitive values—for example, * for an alphabetic string such as a name, or 0 for a numeric string
+                                such as ZIP code or credit card number. This string must have a length of 1. If not supplied, this value defaults to * for
+                                strings, and 0 for digits.
+                            - !ruby/object:Api::Type::Integer
+                              name: 'numberToMask'
+                              description: |
+                                Number of characters to mask. If not set, all matching chars will be masked. Skipped characters do not count towards this tally.
+                                If numberToMask is negative, this denotes inverse masking. Cloud DLP masks all but a number of characters. For example, suppose you have the following values:
+                                - maskingCharacter is *
```

Comment:
could be worth changing the docs for this field to match the Terraform field names - for example `masking_character`, `reverse_order`, etc.

Reply:
Done in [ecf83d6](https://github.com/GoogleCloudPlatform/magic-modules/pull/6872/commits/ecf83d6134db5a328522cc87faa8e623af763377)